File: huawei/test/test_priority_queue.cpp
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

struct Coordinate {
    int x;
    int y;
    int weight;

    Coordinate(int x, int y, int w) : x(x), y(y), weight(w) {}

    // 定义比较函数
    bool operator>(const Coordinate& other) const {
        return weight > other.weight;
    }
};

int main() {
    priority_queue<Coordinate, std::vector<Coordinate>, std::greater<Coordinate>> minHeap;

    // 添加坐标对
    minHeap.push(Coordinate(1, 2, 10));
    minHeap.push(Coordinate(3, 4, 5));
    minHeap.push(Coordinate(5, 6, 8));

    while (!minHeap.empty()) {
        Coordinate top = minHeap.top();
        minHeap.pop();
        std::cout << "Coordinate: (" << top.x << ", " << top.y << "), Weight: " << top.weight << std::endl;
    }

    return 0;
}

File: huawei/2_Transcoding_Network Delivery/test/test_priority_queue.py
import queue

# 创建一个优先队列
q = queue.PriorityQueue()

# 添加元素到队列
q.put((3, 'Apple'))
q.put((1, 'Banana'))
q.put((2, 'Orange'))

# 从队列中获取元素
while not q.empty():
    item = q.get()
    print(item[1])  # 输出元素的值

# 输出结果：
# Banana
# Orange
# Apple